From: Soc Virnyl Estela <contact@uncomfyhalomacro.pl>
Date: Sat, 09 Dec 2023 16:30:31 +0800
Subject: Change --link-static to --link-shared

If we don't change it to shared, it will fail to link it
since openSUSE's llvm does not have static libraries.
---

diff -ruN julia-1.9.4.orig/src/Makefile julia/src/Makefile
--- julia-1.9.4.orig/src/Makefile	2023-12-08 06:25:53.604185583 +0800
+++ julia/src/Makefile	2023-12-08 13:19:45.202318921 +0800
@@ -125,7 +125,7 @@
 # USE_SYSTEM_LLVM == 0
 ifneq ($(USE_LLVM_SHLIB),1)
 # USE_LLVM_SHLIB != 1
-CG_LLVMLINK += $(LLVM_LDFLAGS) $(shell $(LLVM_CONFIG_HOST) --libs $(CG_LLVM_LIBS) --link-static) $($(LLVM_LDFLAGS) $(shell $(LLVM_CONFIG_HOST) --system-libs 2> /dev/null)
+CG_LLVMLINK += $(LLVM_LDFLAGS) $(shell $(LLVM_CONFIG_HOST) --libs $(CG_LLVM_LIBS) --link-shared) $($(LLVM_LDFLAGS) $(shell $(LLVM_CONFIG_HOST) --system-libs 2> /dev/null)
 else
 # USE_LLVM_SHLIB == 1
 ifeq ($(OS), Darwin)
@@ -141,7 +141,7 @@
 endif # USE_LLVM_SHLIB == 1
 endif # JULIACODEGEN == LLVM
 
-RT_LLVM_LINK_ARGS := $(shell $(LLVM_CONFIG_HOST) --libs $(RT_LLVM_LIBS) --system-libs --link-static)
+RT_LLVM_LINK_ARGS := $(shell $(LLVM_CONFIG_HOST) --libs $(RT_LLVM_LIBS) --system-libs --link-shared)
 RT_LLVMLINK += $(LLVM_LDFLAGS) $(RT_LLVM_LINK_ARGS)
 ifeq ($(OS), WINNT)
 RT_LLVMLINK += -luuid -lole32
